jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerSessionContext.kt

```diff
@@ -162,135 +170,249 @@ data class CodeModernizerSessionContext(
+                buildlogBuilder.appendLine("$mavenCommand clean install failed")
+            }
+            return installOutput
+        }
+
+        fun runCopyCommand(mavenCommand: String): ProcessOutput {
+            buildlogBuilder.appendLine("Command Run: $mavenCommand dependency:copy-dependencies")
+            val copyCommand = buildList {
                 add(mavenCommand)
-                addAll(commandList)
+                addAll(copyCommandList)
             }
-            val commandLine = GeneralCommandLine(command)
+            val copyCommandLine = GeneralCommandLine(copyCommand)
                 .withWorkDirectory(sourceFolder)
                 .withRedirectErrorStream(true)
-            val output = ExecUtil.execAndGetOutput(commandLine)
-            return output
+            val copyOutput = ExecUtil.execAndGetOutput(copyCommandLine)
+            buildlogBuilder.appendLine("$copyOutput")
+            if (copyOutput.exitCode == 0) {
+                LOG.info { "$mavenCommand copy-dependencies succeeded" }
+                buildlogBuilder.appendLine("$mavenCommand copy-dependencies succeeded")
+            } else {
+                LOG.error { "$mavenCommand copy-dependencies failed" }
+                buildlogBuilder.appendLine("$mavenCommand copy-dependencies failed")
+            }
+            return copyOutput
         }
 
         // 1. Try to execute Maven Wrapper Command
-        LOG.warn { "Executing ./mvnw" }
         var shouldTryMvnCommand = true
+        val mvnw = if (SystemInfo.isWindows) {
+            "./mvnw.cmd"
+        } else {
+            "./mvnw"
+        }
         try {
-            val mvnw = if (SystemInfo.isWindows) {
-                "./mvnw.cmd"
-            } else {
-                "./mvnw"
+            LOG.info { "Executing $mvnw" }
+            val installOutput = runInstallCommand(mvnw)
+            if (installOutput.exitCode != 0) {
+                LOG.error { "$mvnw install output: $installOutput" }
+                val error = "Maven Install: The exitCode should be 0 while it was ${installOutput.exitCode}"
+                CodetransformTelemetry.mvnBuildFailed(
+                    codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
+                    codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
+                    reason = error
+                )
             }
-            val output = runCommand(mvnw)
-            if (output.exitCode != 0) {
-                LOG.error { "mvnw command output:\n$output" }
-                val error = "The exitCode should be 0 while it was ${output.exitCode}"
+            // TODO: currently running copy dependencies even if install failed.
+            //  This Should be updated to fast fail the transform if install failed.
+            val copyOutput = runCopyCommand(mvnw)
+            if (copyOutput.exitCode != 0) {
+                LOG.error { "$mvnw command output: $copyOutput" }
```

Comment:
nit: make this message here (and below in other `LOG` statements) to clarify that this is the `copy` command being run, and in other places it is the `install` command being run. Will make it easier to look through the logs